components/http_server/include/http_server.h

```diff
@@ -435,6 +507,64 @@ esp_err_t httpd_set_recv_override(httpd_req_t *r, httpd_recv_func_t recv_func);
  */
 esp_err_t httpd_set_send_override(httpd_req_t *r, httpd_send_func_t send_func);
 
+/**
+ * @brief   Override web server's pending function
+ *
+ * This function overrides the web server's pending function. This function is
+ * used to test for pending bytes in a socket.
+ *
+ * @note    This API is supposed to be called only from the context of
+ *          a URI handler where httpd_req_t* request pointer is valid.
+ *
+ * @param[in] r            The request being responded to
+ * @param[in] pending_func The pending function to be set for this request
+ *
+ * @return
+ *  - ESP_OK : On successfully registering override
+ *  - ESP_ERR_INVALID_ARG : Null arguments
+ *  - ESP_ERR_HTTPD_INVALID_REQ : Invalid request pointer
+ */
+esp_err_t httpd_set_pending_override(httpd_req_t *r, httpd_pending_func_t pending_func);
+
+/**
+ * @brief   Override web server's send function (by session FD)
+ *
+ * @see httpd_set_send_override()
+ *
+ * @param[in] hd  HTTPD instance handle
+ * @param[in] fd  session socket FD
+ * @param[in] send_func The send function to be set for this session
+ *
+ * @return status code
+ */
+esp_err_t httpd_set_sess_send_override(httpd_handle_t hd, int sockfd, httpd_send_func_t send_func);
+
+/**
+ * @brief   Override web server's receive function (by session FD)
+ *
+ * @see httpd_set_recv_override()
+ *
+ * @param[in] hd  HTTPD instance handle
+ * @param[in] fd  session socket FD
+ * @param[in] recv_func The receive function to be set for this session
+ *
+ * @return status code
+ */
+esp_err_t httpd_set_sess_recv_override(httpd_handle_t hd, int sockfd, httpd_recv_func_t recv_func);
+
+/**
+ * @brief   Override web server's pending function (by session FD)
+ *
+ * @see httpd_set_pending_override()
+ *
+ * @param[in] hd  HTTPD instance handle
+ * @param[in] fd  session socket FD
+ * @param[in] pending_func The receive function to be set for this session
+ *
+ * @return status code
+ */
+esp_err_t httpd_set_sess_pending_override(httpd_handle_t hd, int sockfd, httpd_pending_func_t pending_func);
+
```

Comment:
new setters were needed to allow using them in the open_fn (where no request exists)